src/BugsnagUnity/Client.cs

```diff
@@ -335,15 +315,16 @@ void Notify(Exception[] exceptions, HandledState handledState, Middleware callba
                 @event.AddAndroidProjectPackagesToEvent(Configuration.ProjectPackages);
             }
 
-            var report = new Report(Configuration, @event);
-
-            lock (MiddlewareLock)
+            lock (CallbackLock)
             {
-                foreach (var middleware in Middleware)
+                foreach (var onErrorCallback in Configuration.GetOnErrorCallbacks())
                 {
                     try
                     {
-                        middleware(report);
+                        if (!onErrorCallback.Invoke(@event))
+                        {
+                            return;
+                        }
                     }
                     catch (System.Exception)
```

Comment:
I think the behaviour detailed in the spec is to log out some details of the error

Reply:
Due to potential loop, logging errors in the Unity sdk is not possible when the issue is part of the notifiy proccess.